Example_Programs/calculator_complex/test/test_complex_calculator.c

```diff
@@ -0,0 +1,132 @@
+#include "complex_calculator.h"
+#include "unity.h"
+
+static complex_t c1 = {0, 0};
+static complex_t c2 = {0, 0};
+static complex_t c3 = {0, 0};
+static complex_t c4 = {0, 0};
+static complex_t result = {0, 0};
+/* Required by the unity test framework */
+void setUp()
+{
+}
+/* Required by the unity test framework */
+void tearDown()
+{
+}
+
+void test_zero(void)
+{
+    // Can ommi the below intialization as it is done at the declaration time
+    c1.real = 0;
+    c1.imaginary = 0;
+    
+    c2.real = 0;
+    c2.imaginary = 0;
+    
+    // Testing sum
+    TEST_ASSERT_EQUAL(SUCCESS, complex_sum(&c1, &c2, &result));
+    TEST_ASSERT_EQUAL(0, result.real);
+    TEST_ASSERT_EQUAL(0, result.imaginary);
+    // Testing subtraction
+    TEST_ASSERT_EQUAL(SUCCESS, complex_sub(&c1, &c2, &result));
```

Comment:
Write different test cases for each function. i.e test function for sum, division, multiply and subtract separately.
Its more readable and also easy to find the bugs.

Reply:
Done, please review if that is what's asked. Thank you Sir.